src/Parsers/ParserCreateQuery.h

```diff
@@ -163,6 +165,36 @@ bool IParserColumnDeclaration<NameParser>::parseImpl(Pos & pos, ASTPtr & node, E
     if (require_type && !type && !default_expression)
         return false; /// reject column name without type
 
+    auto first_word = type->getID();
+
+    if (boost::algorithm::to_lower_copy(first_word) == "function_double") {
```

Comment:
Why not to move it to `ParserIdentifierWithOptionalParameters`? Or may be even make separate `ParserDataType`, because data type parsing becomes too complex with multi-word identifiers. It can be done like the following (it's just a draft and probably will not work or can be done a better way):
```
ASTPtr type;
ParserIdentifierWithOptionalParameters type_parser;
if (!type_parser.parse(pos, type, expected)
    return false;
/// Cast to ASTFunction looks better then using getID()
ASTFunction & type_func = type->as<ASTFunction>();
if (!type.arguments.empty())
{
    /// It cannot be multi-word type name if it has arguments after the first word
    return true;
}

/// Special cases for compatibility with SQL standard. We can parse several words as type name
/// only for certain first words, otherwise we don't know how many words to parse
if (boost::algorithm::iequals(func.name, "DOUBLE"))
{
    if (ParserKeyword{"PRESICION"}.ignore(pos))
    {
        /// "DOUBLE PRESICION" can be registered in factory.
        /// Also the second word will not be lost in formatImpl(...)
        func.name += " PRESICION";
    }
} 
else if (boost::algorithm::iequals(func.name, "CHAR"))
...
///TODO can multi-word name have parameters? 
```